File: uva/uva-10038-jolly-jumpers.py
while True:
    try:
        row = list(map(int, input().split(" ")))
    except:
        break        
    
    n, nums = row[0], row[1:]    
    is_jolly = True
    bit_vec = [False for _ in range(4096)]
    
    for i in range(1, len(nums)):
        c = abs(nums[i] - nums[i-1])
        if (c != 0 and c < n and not bit_vec[c]):
            bit_vec[c] = True
        else:
            is_jolly = False
        #end if
    #end for
    
    if (is_jolly):
        print("Jolly")        
    else:
        print("Not jolly")
        
#end while    

File: uva/uva-10038-jolly-jumpers.cpp
#include <bitset>
#include <iostream>
using namespace std;

const int SIZE = 4096;

int main() {
    
    int n;
    while (cin >> n)
    {
        bitset<SIZE> bitvec;
        int a, b; cin >> b; bool is_jolly = true;
        for (int i = 1; i < n; i++)
        {
            a = b; cin >> b;
            int c = abs(b-a);
            if (c !=0 && c < n && !bitvec[c]) { bitvec.set(c); }
            else { is_jolly = false; }
        }
        is_jolly ? cout << "Jolly" << endl : cout << "Not jolly" << endl;
    }
    return 0;
}